### ISSUE TITLE
meson: how-to `make install`?

### ISSUE THREAD

**Issue by k0ste** [pgbackrest/pgbackrest#2332]
Hi, I try to switch version 2.51 to meson build, after `NOTE TO PACKAGERS`

My current commands is:

```shell
meson setup --buildtype "release" "pgbackrest" "pgbackrest-release-2.51"
ninja -vC "pgbackrest"
meson install -C "pgbackrest" --destdir="/tmp/here"
```

The build complete flawless, on install subcommand meson tells me:

```
ninja: Entering directory `/srv/raid/filez/sandbox/AUR/pgbackrest/src/pgbackrest'
[2/2] Generating src/command/help/help.auto.c.inc with a custom command
Nothing to install.
```

**Comment by pgstef** [comment]
Hi,

Have you looked into the [user-guide](https://pgbackrest.org/user-guide.html#build)?
The user-guide doesn't really use the install step, it just uses a simple copy from `/build/pgbackrest/src/pgbackrest` to `/usr/bin`. Which is also what was done in the old make [section](https://pgbackrest.org/user-guide-rhel.html#build). And which is also what the [rpm pgdg](https://git.postgresql.org/gitweb/?p=pgrpms.git;a=blob_plain;f=rpm/redhat/main/common/pgbackrest/main/pgbackrest.spec;hb=HEAD) package is doing.

What's your OS btw?

I recently updated my Ansible playbook to use meson too:

```yml
- set_fact:
    pkg_config_path: "PKG_CONFIG_PATH=/usr/pgsql-{{ pg_version }}/lib/pkgconfig"
  when: ansible_os_family == 'RedHat'

- name: Configure pgbackrest
  shell: "{{ pkg_config_path | default('', true) }} meson setup {{ pgbackrest_build_dir }} {{ pgbackrest_src_dir }} --prefix={{ pgbackrest_build_prefix }} --bindir={{ pgbackrest_build_prefix }}/bin"

- name: Build pgbackrest
  shell: "ninja -C {{ pgbackrest_build_dir }}"

- name: Install pgbackrest
  shell: "ninja install"
  args:
    chdir: "{{ pgbackrest_build_dir }}"
```

which pretty much would give on rhel systems:

```bash
export PKG_CONFIG_PATH=/usr/pgsql-16/lib/pkgconfig 
meson setup $pgbackrest_build_dir $pgbackrest_src_dir --prefix=/usr --bindir=/usr/bin
cd $pgbackrest_build_dir
ninja
su
ninja install
```

**Comment by k0ste** [comment]
```
- name: Install pgbackrest
  shell: "ninja install"
  args:
    chdir: "{{ pgbackrest_build_dir }}"
```

This command will produce this output

```
[1/3] /srv/raid/filez/sandbox/AUR/pgbackrest/src/pgbackrest/src/build-code postgres ../pgbackrest-release-2.51/src/postgres/../..
[2/3] /srv/raid/filez/sandbox/AUR/pgbackrest/src/pgbackrest/src/build-code help ../pgbackrest-release-2.51/src/command/help/../../..
[2/3] /usr/bin/meson install --no-rebuild
Nothing to install.
```

**Comment by k0ste** [comment]
Never mind, the fix in #2312

```shell
ninja: Entering directory `/srv/raid/filez/sandbox/AUR/pgbackrest/src/build'
[2/2] Generating src/command/help/help.auto.c.inc with a custom command
Installing src/pgbackrest to /tmp/kek/usr/bin
```

**Comment by dwsteele** [comment]
> Never mind, the fix in https://github.com/pgbackrest/pgbackrest/pull/2312

Sorry for the confusion here. If you are a packager, might be easier to wait for 2.52 before using the meson build. We'll keep the autoconf/make build around until we are confident the meson build works well.

**Comment by k0ste** [comment]
> Sorry for the confusion here. If you are a packager, might be easier to wait for 2.52 before using the meson build. We'll keep the autoconf/make build around until we are confident the meson build works well.

Thank you, but a _little_ "but": the release immediately screams that "dude, it’s time!"

<img width="1009" alt="Screenshot 2024-04-20 at 11 24 55" src="https://github.com/pgbackrest/pgbackrest/assets/7759548/9183d48c-1d09-4087-aae2-f7bf13cabdca">

Therefore, we have already adapted the [AUR build](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=pgbackrest#n22) to the meson. In my opinion, it would be right to immediately release hot-fix release `2.51.1` to fix this. It doesn't break anything and is easy to do, well at least it seems so to me 🙂

**Comment by dwsteele** [comment]
I don't think a new release is a good idea -- we don't really have an established hotfix system so it is likely to cause confusion. However, what I have done is remove the note and add it to the next release. So hopefully no new packagers will start the meson transition on 2.51.

A couple of comments on the package:

1) We don't have a dependency on xz and given the bad odor around that project recently I'd just as soon we don't have a dependency on it.

2) Perl should not be required as a build dependency anymore unless you are building the docs (which you do not appear to be doing).

3) Also not sure why icu is a dependency, but maybe that is a Arch thing?

4) I find the patch here (https://aur.archlinux.org/cgit/aur.git/tree/pgbackrest-2312.patch?h=pgbackrest) more complicated than it needs to be. Maybe it would be better to use the one from Debian: https://salsa.debian.org/postgresql/pgbackrest/-/blob/master/debian/patches/meson-install.patch?ref_type=heads?

And again, sorry for the trouble.
